src/main/java/net/pms/formats/FormatFactory.java

```diff
@@ -148,7 +147,24 @@ public static Format getAssociatedFormat(final String filename) {
 			formatsLock.readLock().unlock();
 		}
 
-		LOGGER.trace("Could not match any format to \"" + filename + "\"");
+		LOGGER.trace("Could not match any format to \"{}\"", filename);
+		return null;
+	}
+
+	public static Format getFormat(Class<? extends Format> clazz) {
+		if (clazz == null) {
+			return null;
+		}
+		formatsLock.readLock().lock();
+		try {
+			for (Format format : formats) {
+				if (format.getClass().equals(clazz)) {
```

Comment:
Aaah..i didn't think about that. Thank you, i'll go sleep smarter today ;)

P.S. I hope that it was benign at the hospital. That kind of things that can happen when we try to handle MEncoder :wink: